### ISSUE TITLE
Access control broken under RC6

### ISSUE THREAD

**Issue by PeterSchumacher** [Luracast/Restler#420]
When entering '12345' (user), all functions from the Access class is shown in explorer - including the admin one.
If admin function is accessed under api_key '12345', it - correctly- reports an error.
So it's probably just a change in Resources that generates faulty json for explorer.
I set a breakpoint in the 'verifyAccess' static function and it is never reached.

BTW, I'm using "restler/framework": "3.0.0-RC6" in my composer.json.

**Comment by stevenbalthazor** [comment]
Hello - I think I found a fix for a similar (if not the same issue).

I was trying to get api calls to respect the access control as specified via phpdoc (like @class AccessControl {@requires user} ).  The explorer was respecting this access control requirement, but if I made an api call to the endpoint, the access control was not being respected.

Current code in Restler.php (about line 958):

    } else {
        $this->authenticated = true;
    }

I think the following replacement code solves the problem:

    } else {
        if (Defaults::$accessControlFunction && (!call_user_func(Defaults::$accessControlFunction, $o->metadata))
            ) {
                    throw new RestException(401);
        }else {
                    $this->authenticated = true;
        }
    }

Regards,
Steven Balthazor
